xbmc/interfaces/json-rpc/InputOperations.cpp

```diff
@@ -84,6 +88,45 @@ JSONRPC_STATUS CInputOperations::ExecuteAction(const std::string &method, ITrans
   return SendAction(action);
 }
 
+JSONRPC_STATUS CInputOperations::Button(const std::string &method, ITransportLayer *transport, IClient *client, const CVariant &parameterObject, CVariant &result)
+{
+  std::string button = parameterObject["button"].asString();
+  std::string keymap = parameterObject["keymap"].asString();
+
+  int keycode = 0;
+  if ( keymap.compare("KB") == 0 ) // standard keyboard map
+  {
+  keycode = CKeyboardTranslator::TranslateString( button.c_str() );
+  }
+  else if  ( keymap.compare("XG") == 0 ) // xbox gamepad map
+  {
+  keycode = CGamepadTranslator::TranslateString( button.c_str() );
+  }
+  else if  ( keymap.compare("R1") == 0 ) // xbox remote map
+  {
+  keycode = CIRTranslator::TranslateString( button.c_str() );
+  }
+  else if  ( keymap.compare("R2") == 0 ) // xbox universal remote map
+  {
+  keycode = CIRTranslator::TranslateUniversalRemoteString( button.c_str() );
+  }
+  else
+  {
+    return InvalidParams;
+  }
+  
+  XBMC_Event newEvent;
+  newEvent.type = XBMC_BUTTON;
+  newEvent.keybutton.button = keycode;
+  newEvent.keybutton.holdtime = 0;
+  
+  CInputManager &inputmanager = CServiceBroker::GetInputManager();
```

Comment:
> Left-align `*` and `&` to the base type they modify.

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#52-pointer-and-reference-types